sae_lens/training/sae_group.py

```diff
@@ -35,32 +35,47 @@ def _init_autoencoders(self, cfg: LanguageModelSAERunnerConfig):
 
         # Create all combinations of hyperparameters
         for combination in product(*values):
+
             params = dict(zip(keys, combination))
             cfg_copy = dataclasses.replace(cfg, **params)
+
             cfg_copy.__post_init__()
             # Insert the layer into the hookpoint
             cfg_copy.hook_point = cfg_copy.hook_point.format(
                 layer=cfg_copy.hook_point_layer
             )
+
+            sae = SparseAutoencoder(cfg_copy)
+
+            sae_name = (
+                f"{sae.cfg.model_name}_{sae.cfg.hook_point}_{sae.cfg.d_sae}_"
+                + "_".join([f"{k}_{v}" for k, v in zip(keys, combination)])
+            )
+
             # Create and store both the SparseAutoencoder instance and its parameters
-            self.autoencoders.append(SparseAutoencoder(cfg_copy))
+            self.autoencoders[sae_name] = sae
+
+    def __getitem__(self, key: str) -> SparseAutoencoder:
+        return self.autoencoders[key]
 
-    def __iter__(self) -> Iterator[SparseAutoencoder]:
+    def __iter__(self) -> Iterator[tuple[str, SparseAutoencoder]]:
```

Comment:
just want to flag that for Python mappings (like dictionaries), the `__iter__` behavior is to just iterate the keys, not keys and values. More info: [https://docs.python.org/3/reference/datamodel.html?emulating-container-types](https://docs.python.org/3/reference/datamodel.html?emulating-container-types=#object.__iter__). Not actually a huge deal, since we're not implementing most of the core mapping methods like `__getitem__`, but just flagging in case we want this class to work like a real Python dict at some point.

Reply:
I'd be up for fixing this in the near future.